neo/Ledger/BlockChain.State.cs

```diff
@@ -54,20 +54,18 @@ private RelayResultReason OnNewStateRoot(StateRoot stateRoot)
                 stateRootCache.Add(stateRoot.Index, stateRoot);
                 return RelayResultReason.Succeed;
             }
-            using (Snapshot snapshot = GetSnapshot())
+
+            while (stateRoot.Index <= Height)
             {
-                while (stateRoot.Index <= Height)
+                stateRootCache.Remove(stateRoot.Index);
```

Comment:
Why remove the state before verify?

Reply:
I only remove the un-use snapshot:
```
using( Snapshot snapshot = GetSnapshot())
{
    // the while region
}
```
I didn't change any logic in the while statement.